lib/openstax/aws/stack.rb

```diff
@@ -1,5 +1,78 @@
+require 'json'
+
 module OpenStax::Aws
   class Stack
+    class Event
+      def initialize(aws_stack_event)
+        @aws_stack_event = aws_stack_event
+      end
+            
+      def status
+        @aws_stack_event.data.resource_status
+      end
+            
+      def reason
+        @aws_stack_event.data.resource_status_reason
+      end
+            
+      def failure?
+        %w(
+          ROLLBACK_COMPLETE
+          ROLLBACK_IN_PROGRESS
+          CREATE_FAILED
+          ROLLBACK_FAILED
+          DELETE_FAILED
+          UPDATE_ROLLBACK_FAILED
+          IMPORT_ROLLBACK_FAILED
+        ).include?(status)
+      end
+            
+      def user_initiated?
+        reason == "User Initiated"
+      end
+    end
+
+    class Status
+      def initialize(stack)
+        @stack = stack
+      end
+
+      def status_text
+        @stack.aws_stack.stack_status
+      end
+
+      def failure?
+        %w(
+          ROLLBACK_COMPLETE
+          ROLLBACK_IN_PROGRESS
+          CREATE_FAILED
+          ROLLBACK_FAILED
+          DELETE_FAILED
+          UPDATE_ROLLBACK_FAILED
+          IMPORT_ROLLBACK_FAILED
+        ).include?(status_text)
+      end
+
+      def failed_events_since_last_user_event
+        @stack.events.each_with_object([]) do |aws_stack_event, array|
+          event = Event.new(aws_stack_event)
+          array.push(event) if (event.failure? && status_text)
+          return array if event.user_initiated?
+        end
+      end
+
+      def to_h
+        {
+          name: @stack.name,
```

Comment:
Is the name still needed here?

Reply:
maybe? I thought it was important no have it, since I don't have a template or whatever to base on what we actually want to see from this entire PR

Reply:
see if you need it from the unified-deployment PR.  You already have the name now in https://github.com/openstax/aws-ruby/pull/62/files#diff-2239543d83a02cdb632c49b9ccd2e5ebR10